src/index.ts

```diff
@@ -94,6 +95,13 @@ export interface UploadConfig {
    */
   configPath?: string;
 
+  /**
+   * For each API request we send, you may specify custom request options that
+   * we'll add onto the request. The request options follow the node-fetch API:
+   * https://github.com/node-fetch/node-fetch#options.
+   */
+  customRequestOptions?: object;
```

Comment:
`object` doesn't seem right.  If this is in fact using gaxios, `GaxiosOptions` is probably what we're looking for.  If it really is fetch, let's bring in `@types/node-fetch`